inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -623,10 +642,10 @@ void GNAGraphCompiler::finalizeConvolution2DPrimitive(InferenceEngine::CNNLayerP
     // TODO: Jira: 43659 - the issue also appears when after conv there's an eltwise or activation
     // For last layer or when next ones are only non functional, the data can be reordered when exporting scores
     // For other cases inserting permute is required if data are reordered
-    auto isNonFunctional = [](CNNLayerPtr l) {
-        return LayerInfo(l).isNonFunctional();
+    auto keepsLayout = [](CNNLayerPtr l) {
```

Comment:
I suppose pooling can decrease a tensor size, so num_rotate_output_rows calculated on the base of convolution outputs will be incorrect.
At the same time I'm not sure if filling of rotation parameters is actually needed here. Now it's handled in gna_plugin.cpp and covers cases when there are any other layers after convolution, so I removed this code from 1d convolution primitive handling. Is there some special reason to keep it for 2d convolution?

Reply:
You were right it is not needed. Fixed.